ERCS/erc-7811.md

```diff
@@ -0,0 +1,239 @@
+
+`assetTypeFilter` is an **OPTIONAL** field that specifies an array of asset types, as defined in this ERC. If `assetTypeFilter` field is provided, wallet **MUST** include only those assets in the response.
+
+`chainFilter` is an **OPTIONAL** field that specifies an array of chain ids, where each value in the array **MUST** be a valid [EIP-155](./eip-155.md) chainId
+
+#### Example request
+
+```json
+{
+  "account": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
+  "assetFilter": {
+    "0x1": ["0x456", "native"],
+    "0xa": ["0x789", "native"]
+  },
+  "assetTypeFilter": ["ERC20", "native"],
+  "chainFilter": ["0x1"]
+}
+```
+
+#### Response schema
+
+```ts
+type Asset = {
+  address: Hex | "native";
+  balance: Hex;
+  type: string;
+  metadata: any;
+};
+type WalletGetAssetsResponse = Record<Hex, Asset[]>;
+```
+
+The key **SHOULD** be [EIP-155](./eip-155.md) chainId
+
+Asset fields:
+
+`address` is the address of the asset as `Hex` or `native` string for native assets.
+
+`balance` is the balance of the asset as `Hex`
+
+**`type`:** A string indicating the type of the asset. Common asset types include but **aren’t limited to**:
+
+- **`ERC20`:** For [ERC-20](./eip-20.md) tokens
+- **`ERC721`:** For [ERC-721](./eip-721.md) tokens (NFTs)
+- **`native`:** For the chain's native asset
+
+**`metadata`:** An **OPTIONAL** object containing additional information about the asset. The specific fields within the metadata object may vary depending on the asset type and the wallet's implementation.
+
+#### Example response
+
+```json
+{
+  "0x1": [
+    {
+      "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+      "balance": "0xcaaea35047fe5702",
+      "type": "ERC20",
+      "metadata": {
+        "name": "Token",
+        "symbol": "TOK",
+        "decimals": 18
+      }
+    },
+    {
+      "address": "native",
+      "balance": "0xcaaea35047fe5702",
+      "type": "native"
+    }
+  ],
+  "0xa": [
+    {
+      "address": "0x456",
+      "balance": "0xcd5595",
+      "type": "ERC721",
+      "metadata": {
+        //...
+      }
+    }
+  ]
+}
+```
+
+### Asset definitions
+
+This section specifies a structure for the most commonly used asset types.
+This ERC does not specify an exhaustive list of asset types.
+Since the type is a generic string, there could be a mismatch between the type Dapp expects and the one returned by the wallet.
+It’s important that no two assets share the same type.
+Therefore, new asset types should be specified in future ERCs.
+
+**Native**
+
+```ts
+type NativeAsset = {
+  address: "native";
+  balance: Hex;
+  type: "native";
+};
+```
+
+Example:
+
+```json
+{
+  "address": "native",
+  "balance": "0xcaaea35047fe5702",
+  "type": "native"
+}
+```
+
+**ERC-20 Token**
+
+```ts
+type Erc20Asset = {
+  address: Hex;
+  balance: Hex;
+  type: "ERC20";
+  metadata: {
+    name: string;
+    symbol: string;
+    decimals: number;
```

Comment:
TIL technically these are optional fields. Shouldn't we follow that? https://github.com/ethereum/ERCs/blob/master/ERCS/erc-20.md#token